packages/nimble-components/src/utilities/type-extensions.d.ts

```diff
@@ -0,0 +1,17 @@
+/**
+ * This is a workaround for an incomplete definition of the native dialog element. Remove when using Typescript >=4.8.3.
+ * https://github.com/microsoft/TypeScript/issues/48267
+ * @internal
+ */
+interface HTMLDialogElement {
+    showModal(): void;
+    close(): void;
+}
```

Comment:
Can we make this align with the TS definitions (https://github.com/microsoft/TypeScript-DOM-lib-generator/blob/27fa26c78ce2401310e26a13ea233323707d88c2/baselines/dom.generated.d.ts#L10012-L10037)

```
showModal(): void;
close(returnValue?: string): void;
```